lib/src/run.dart

```diff
@@ -144,3 +152,56 @@ stderr:
 
 $stderr""";
 }
+
+
+/// Arguments passed to [Process.run] .
+/// See [Process.run] for more details.
+class RunOptions {
+  final String workingDirectory;
+  final Map<String, String> environment;
+  final bool includeParentEnvironment;
+  final bool runInShell;
+  final Encoding stdoutEncoding;
+  final Encoding stderrEncoding;
+
+  RunOptions({this.workingDirectory, this.environment,
+      this.includeParentEnvironment: true, this.runInShell: false,
+      this.stdoutEncoding: SYSTEM_ENCODING,
+      this.stderrEncoding: SYSTEM_ENCODING});
+
+  /// Create a clone when it's necessary to modify the passed runOptions to
+  /// avoid modifying the argument.
+  RunOptions clone() {
```

Comment:
It'd be really nice to keep the fields final.  What about:

(note the Map copy)

```
  RunOptions withOptions({String workingDirectory, Map<String, String> environment,
      bool includeParentEnvironment, bool runInShell,
      Encoding stdoutEncoding,
      Encoding stderrEncoding}) => new RunOptions(
      workingDirectory: workingDirectory ?? this.workingDirectory,
      environment: environment ?? new Map.from(this.environment),
      includeParentEnvironment: includeParentEnvironment ?? this.includeParentEnvironment,
      stdoutEncoding: stdoutEncoding ?? this.stdoutEncoding,
      stderrEncoding: stderrEncoding ?? this.stderrEncoding
);
```

Then you can still do:

`runOptions.withOptions()..environment['xxx']='yyy';`

but also:

`runOptions.withOptions(workingDirectory: 'x/y/z');`

The name `clone` or `copy` might work too.